--- origsrc/mpg123-1.15.4/src/libmpg123/mangle.h	2013-05-20 06:39:33.000000000 -0500
+++ src/mpg123-1.15.4/src/libmpg123/mangle.h	2013-09-17 21:14:06.786802400 -0500
@@ -58,7 +58,7 @@
 #if defined(__USER_LABEL_PREFIX__)
 #define ASM_NAME(a) MANGLE_MACROCAT(__USER_LABEL_PREFIX__,a)
 #define ASM_VALUE(a) MANGLE_MACROCAT($,ASM_NAME(a))
-#elif defined(__CYGWIN__) || defined(_WIN32) && !defined (_WIN64) || defined(__OS2__) || \
+#elif ((defined(__CYGWIN__) || defined(_WIN32)) && !defined (__x86_64__)) || defined(__OS2__) || \
    (defined(__OpenBSD__) && !defined(__ELF__)) || defined(__APPLE__)
 #define ASM_NAME(a) MANGLE_MACROCAT(_,a)
 #define ASM_VALUE(a) MANGLE_MACROCAT($_,a)
